js/memory.js

```diff
@@ -1,9 +1,37 @@
 class MemoryGame {
-  constructor(cards){
+  constructor(cards) {
     this.cards = cards;
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
     // add the rest of the class properties here
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
+  shuffleCards() {
+    for (let i = this.cards.length - 1; i > 0; i--) {
+      let j = Math.floor(Math.random() * (i + 1));
+      let temp = this.cards[i];
+      this.cards[i] = this.cards[j];
+      this.cards[j] = temp;
+    }
+  }
+
+
+  checkIfPair(card1, card2) {
+
+    this.pairsClicked++
+
+    if (card1 === card2) {
+      this.pairsGuessed++
+      return true;
+    } else {
+      return false;
+    }
+  }
+  isFinished() {
+    if (this.pairsGuessed === 8) {
```

Comment:
```suggestion
    if (this.pairsGuessed === this.cards.length / 2) {
```
Me gustó la manera como resolviste este atasco, consultando en las "tripas" de Jasmine, esa es la idea.

Pero si te das cuenta, si que lo estabas contemplando bien, solo que debías dividir entre 2 pero la longitud del array de cartas totales.